src/mappings/rewardsDistributor.ts

```diff
@@ -90,7 +94,33 @@ export async function handleRewardsDistributed(
       );
       await Delegation.remove(delegator.id);
     }
+
+    if (rewards.gt(0)) {
+      const eraReward = await updateEraReward({
+        indexerId: indexer,
+        delegatorId: delegator.delegatorId,
+        eraId: eraIdx.toHexString(),
+        isCommission: false,
+        claimed: false,
+        amount: rewards.toBigInt() - rewardOld,
+        createdBlock: event.blockNumber,
+        createdTimestamp: biToDate(event.block.timestamp),
+      });
+      accumulatedEraRewards += eraReward.amount;
+    }
   }
+
+  await updateEraReward({
```

Comment:
if `rewards` can be zero, then commission also can be. You probably will want to skip that as well.